pgmpy/models/MarkovChain.py

```diff
@@ -0,0 +1,341 @@
+    set_start_state(state)
+    add_variable(variable, cardinality)
+    add_variables_from(vars_list, cards_list)
+    add_transition_model(variable, transition_dict)
+    sample(start_state, size)
+
+    Examples:
+    ---------
+    Create an empty Markov Chain:
+    >>> from pgmpy.models import MarkovChain as MC
+    >>> model = MC()
+
+    And then add variables to it
+    >>> model.add_variables_from(['intel', 'diff'], [2, 3])
+
+    Or directly create a Markov Chain from a list of variables and their cardinalities
+    >>> model = MC(['intel', 'diff'], [2, 3])
+
+    Add transition models
+    >>> intel_tm = {0: {0: 0.25, 1: 0.75}, 1: {0: 0.5, 1: 0.5}}
+    >>> model.add_transition_model('intel', intel_tm)
+    >>> diff_tm = {0: {0: 0.1, 1: 0.5, 2: 0.4}, 1: {0: 0.2, 1: 0.2, 2: 0.6 }, 2: {0: 0.7, 1: 0.15, 2: 0.15}}
+    >>> model.add_transition_model('diff', diff_tm)
+
+    Set a start state
+    >>> from pgmpy.factors import State
+    >>> model.set_start_state([State('intel': 0), State('diff': 2)])
+
+    Sample from it
+    >>> model.sample(size=5)
+       intel  diff
+    0      0     2
+    1      1     0
+    2      0     1
+    3      1     0
+    4      0     2
+    """
+    def __init__(self, variables=None, card=None, start_state=None):
+        """
+        Parameters:
+        -----------
+        variables: array-like iterable object
+            A list of variables of the model.
+
+        card: array-like iterable object
+            A list of cardinalities of the variables.
+
+        start_state: array-like iterable object
+            List of tuples representing the starting states of the variables.
+        """
+        if variables is None:
+            variables = []
+        if card is None:
+            card = []
+        if not hasattr(variables, '__iter__'):
+            raise ValueError('variables must be an iterable.')
+        if not hasattr(card, '__iter__'):
+            raise ValueError('card must be an iterable.')
+        self.variables = variables
+        self.cardinalities = {v: c for v, c in zip(variables, card)}
+        self.transition_models = {var: {} for var in variables}
+        if start_state is None or self._check_state(start_state):
+            self.state = start_state
+
+    def set_start_state(self, start_state):
+        """
+        Set the start state of the Markov Chain. If the start_state is given as a array-like iterable, its contents
+        are reordered in the internal representation.
+
+        Parameters:
+        -----------
+        start_state: dict or array-like iterable object
+            Dict (or list) of tuples representing the starting states of the variables.
+
+        Examples:
+        ---------
+        >>> from pgmpy.models import MarkovChain as MC
+        >>> from pgmpy.factors import State
+        >>> model = MC(['a', 'b'], [2, 2])
+        >>> model.set_start_state([State('a', 0), State('b', 1)])
+        """
+        if isinstance(start_state, dict):
```

Comment:
Don't accept multiple types. Accept just the array of `State`. We will be making this change in the whole library.